lib/chef/resource/windows_task.rb

```diff
@@ -0,0 +1,237 @@
+                                       :weekly,
+                                       :monthly,
+                                       :once,
+                                       :on_logon,
+                                       :onstart,
+                                       :on_idle], default: :hourly
+      property :start_day, String
+      property :start_time, String
+      property :day, [String, Integer]
+      property :months, String
+      property :idle_time, Integer
+      property :random_delay, String
+      property :execution_time_limit
+
+      attr_accessor :exists, :status, :enabled
+
+      def after_created
+        if random_delay
+          validate_random_delay(random_delay, frequency)
+          duration = sec_to_dur(random_delay)
+          random_delay(duration)
+        end
+
+        if execution_time_limit
+          raise ArgumentError, "Invalid value passed for `execution_time_limit`. Please pass seconds as a String e.g. '60'." if execution_time_limit.to_i == 0
+          duration = sec_to_dur(execution_time_limit)
+          execution_time_limit(duration)
+        else
+          # schtask sets execution_time_limit as PT72H by default
+          # We are setting the default value here so that we can do idempotency check later
+          # Note: We can't use `default` in the property
+          # because it will raise error for Invalid values passed as "PT72H" is not in seconds
+          execution_time_limit("PT72H")
+        end
+
+        validate_start_time(start_time) if frequency == :once
+        validate_start_day(start_day, frequency) if start_day
+        validate_user_and_password(user,password)
+        validate_interactive_setting(interactive_enabled, password)
+        validate_create_frequency_modifier(frequency, frequency_modifier)
+        validate_create_day(day, frequency) if day
+        validate_create_months(months, frequency) if months
+        validate_idle_time(idle_time, frequency) if idle_time
+      end
+
+      private
+
+      def validate_random_delay(random_delay, frequency)
+        if [:once, :on_logon, :onstart, :on_idle].include? frequency
+          raise ArgumentError, "`random_delay` property is supported only for frequency :minute, :hourly, :daily, :weekly and :monthly"
+        end
+
+        raise ArgumentError, "Invalid value passed for `random_delay`. Please pass seconds as a String e.g. '60'." if random_delay.to_i == 0
+      end
+
+      def validate_start_day(start_day, frequency)
+        if [:once, :on_logon, :onstart, :on_idle].include? frequency
+          raise ArgumentError, "`start_day` property is not supported with frequency: #{frequency}"
+        end
+      end
+
+      def validate_start_time(start_time)
+        raise ArgumentError,"`start_time` needs to be provided with `frequency :once`" unless start_time
+      end
+
+      SYSTEM_USERS = ['NT AUTHORITY\SYSTEM', 'SYSTEM', 'NT AUTHORITY\LOCALSERVICE', 'NT AUTHORITY\NETWORKSERVICE'].freeze

```

Comment:
We should add `BUILTIN\USERS` and `USERS` per https://github.com/chef-cookbooks/windows/pull/457 as this allows users to schedule tasks that can run for the currently logged on user.  Helpful when managing client systems or embedded devices.